pystac_client/client.py

```diff
@@ -296,10 +383,17 @@ def get_items(self) -> Iterator["Item_Type"]:
             Iterator[Item]:: Iterator of items whose parent is this
                 catalog.
         """
-        if self._conforms_to(ConformanceClasses.ITEM_SEARCH):
+        if self.conforms_to(ConformanceClasses.ITEM_SEARCH):
             search = self.search()
             yield from search.items()
         else:
+            if self.has_conforms_to():
+                warnings.warn(
+                    DOES_NOT_CONFORM_TO("ITEM_SEARCH"),
+                    category=DoesNotConformTo,
+                    stacklevel=2,
+                )
+            warnings.warn(FALLBACK_MSG, category=FallbackToPystac, stacklevel=2)
```

Comment:
This pattern (check conforms to, maybe do a `DoesNotConformTo`, always do a `FallbackToPystac`) feels like it could be abstracted to its own function? I know we're hanging a lot of specific-use private methods on here, but I just worry about the fragility of this duplicated logic.